src/components/ha-sidebar.html

```diff
@@ -102,19 +102,19 @@
     <paper-menu attr-for-selected='data-panel' selected='[[hass.currentPanel]]' on-iron-select='menuSelect'>
       <paper-icon-item on-tap='menuClicked' data-panel='states'>
         <iron-icon item-icon icon='mdi:apps'></iron-icon>
-        <span class='item-text'>States</span>
+        <span class='item-text'>[[tr(pg, "panel.states")]]</span>
       </paper-icon-item>
 
       <template is='dom-repeat' items='[[panels]]'>
         <paper-icon-item on-tap='menuClicked' data-panel$='[[item.url_path]]'>
           <iron-icon item-icon icon='[[item.icon]]'></iron-icon>
-          <span class='item-text'>[[item.title]]</span>
+          <span class='item-text'>[[tr(pg, item.title)]]</span>
```

Comment:
Lucky pick :)

My first thought would be to expose an API on the HTTP component that allows custom components to register additional translation resources. Then the HTTP component can roll up the extra resources from all custom components and serve them to the client as a json file. Then the frontend can extend its definitions to include the additional translations. (In this system it would be [polyglot.extend](https://github.com/airbnb/polyglot.js#translation). Other APIs should have similar functionality.)